File: java-code/src/main/java/refactoring/introduce_parameter_object/IntroduceParameterObject.java
package refactoring.introduce_parameter_object;

import lombok.extern.slf4j.Slf4j;

@Slf4j
public class IntroduceParameterObject {

  public void readingsOutsideRange(Station station, int min, int max) {
    station.getReadings().stream()
        .filter(r -> r.getTemp() < min || r.getTemp() > max)
        .forEach(r -> log.info("Reading outside range: " + r.getTemp()));
  }
}

File: go-code/introduce_parameter_object/introduce_parameter_object.go
package introduce_parameter_object

import "fmt"

type Reading struct {
	Temp int
	Time string
}

type Station struct {
	Name     string
	Readings []Reading
}

func readingsOutsideRange(station Station, min int, max int) {
	for _, r := range station.Readings {
		if r.Temp < min || r.Temp > max {
			fmt.Println("Reading outside range: ", r.Temp)
		}
	}
}
